src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -264,39 +264,41 @@ const HierarchyList = ({
   const delayedSearch = useCallback(debounce(textInput => handleSearch(textInput), 150), [items]);
 
   return (
-    <List
-      title={title}
-      buttons={buttons}
-      search={
-        hasSearch
-          ? {
-              value: searchValue,
-              onChange: evt => {
-                setSearchValue(evt.target.value);
-                delayedSearch(evt.target.value);
-              },
-            }
-          : null
-      }
-      items={pageSize != null ? itemsToShow : filteredItems}
-      expandedIds={expandedIds}
-      toggleExpansion={id => {
-        if (expandedIds.filter(rowId => rowId === id).length > 0) {
-          // remove id from array
-          setExpandedIds(expandedIds.filter(rowId => rowId !== id));
-        } else {
-          setExpandedIds(expandedIds.concat([id]));
+    <div data-floating-menu-container style={{ position: 'relative' }}>
```

Comment:
This style should be in the .scss file and applied via a class. Do we need the data attribute for something?

Reply:
I will extract to scss.

The data attribute forces the Menu to be in the DOM as a child of what is rendering it. Currently the implementation of this in Carbon has it as a portal (https://reactjs.org/docs/portals.html) which places it outside of our tree of components. I have updated the description of this PR to have the relevant Carbon issue that had reference to this fix.